LanguageFeatures/Patterns/Exhaustiveness/lifting_cast_pattern_A02_t01.dart

```diff
@@ -5,31 +5,35 @@
 /// @assertion The lifted space union for a pattern with matched value type M is
 /// ...
 /// Cast pattern:
-/// The space union spaces for a cast pattern with cast type C is a union of:
-/// - The lifted space union of the cast's subpattern in context C.
-/// - For each space E in the expanded spaces of M:
-///   a. If E is not a subset of C and C is not a subset of M, then the lifted
-///   space union of E.
+/// ...
+/// Let S be the lifted space union of the cast's subpattern in context C.
+/// i. If C is a subset (see below about type subsetting) of S then the result
+///   spaces is the lifted space union of M.
+/// ii. Otherwise, the result spaces is S plus the lifted space union of Null
+///   when C is a non-nullable type, and spaces is S when C is potentially
+///   nullable.
 ///
 /// @description Check a lifted space of a cast pattern in case of a sealed type
 /// @author sgrekhov22@gmail.com
-/// @issue 51877
-
-import "../../../Utils/expect.dart";
 
 sealed class A {
   final int field;
   A(this.field);
 }
+
 class B extends A {
   B(int field) : super(field);
 }
+
 class C extends A {
   C(int field) : super(field);
 }
 
 test1(A a) {
-  switch (a) {
+  switch (a) { // B is not matched
```

Comment:
Perhaps:
```suggestion
  switch (a) { // Not exhaustive, cannot express that "the complement of `C`" will throw.
```
If the subpattern had been guaranteed to accept all objects passing `as C` then the first case would have covered everything. But since the subpattern will reject some instances of `C`, we don't have that, and then the switch isn't exhaustive.

It's confusing to say that `B` is not matched because we could have an object that `implements B, C`, and that one would be matched.

Reply:
Analyzer error message here says "The type 'A' is not exhaustively matched by the switch cases since it doesn't match 'B()'." That's what I'm trying to express in this comment. So, may be "`B()` is not matched here"?

Reply:
But it is not true that if an object is typable as `B` then it will definitely not be matched, it could be typable as `B` as well as `C`, and then it matches one of the two cases just fine.

It is also untrue that an instance of `B` (that is, the value of an instance creation like `B()`, which is an object that is typable as `B`, and not typable as `C`) will not be handled: It will be subject to the type check `as C`, and that operation will throw.

So what are they actually saying? I'm not sure they saying anything about `B`, or at least not anything which is obviously correct.

I think it's safer to say that the rules can't prove exhaustiveness for this switch. The reason why we get such an imprecise result in this case is that the exhaustiveness analysis cannot represent the complement of a set of objects (like "all objects that are not typable as `C`"), and then we only discover that objects matching `C(field: 0)` are handled, and we completely ignore the fact that some objects are handled because `as C` throws.

Perhaps we should just omit that comment, because it gets somewhat verbose if we try to describe the situation. ;-)